src/main/java/suppliers/FigureSupplier.java

```diff
@@ -0,0 +1,46 @@
+package suppliers;
+
+import core.basesyntax.model.Circle;
+import core.basesyntax.model.Figure;
+import core.basesyntax.model.IsoscelesTrapezoid;
+import core.basesyntax.model.Rectangle;
+import core.basesyntax.model.RightTriangle;
+import core.basesyntax.model.Square;
+import java.util.Random;
+
+public class FigureSupplier {
+    private ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        int index = new Random().nextInt(4);
+        switch (index) {
+            case 0:
+                return new Circle(colorSupplier.getRandomColor(), getRandomInt());
+            case 1:
+                return new IsoscelesTrapezoid(colorSupplier.getRandomColor(),
+                        getRandomDouble(), getRandomDouble(), getRandomDouble());
+            case 2:
+                return new Rectangle(colorSupplier.getRandomColor(), getRandomDouble(),
+                        getRandomDouble());
+            case 3:
+                return new RightTriangle(colorSupplier.getRandomColor(), getRandomInt(),
+                        getRandomInt());
+            case 4:
+                return new Square(colorSupplier.getRandomColor(), getRandomDouble());
+            default:
+                return getDefaultFigure();
```

Comment:
We can create 5 private methods getRandomCircle(), getRandomSquare(), ... and call them in switch-case construction, this will increase code readability (Remember private methods should follow after public as helper ones)

Reply:
Done